module.exports = (fn) => {
  // מקבלת ארגומנט יחיד שהוא פנ' אסינכרונית
  return (req, res, next) => {
    //מחזיר פונקציה עם שלושה פרמטרים
    fn(req, res, next).catch(next); // קוראת לפונקציה המקורית שקיבלה ומנסה לתפוס את כל השגיאות שבהוהפונקציה הבאה תיקרא עם השגיאה כארגומנט שלה
  };
};
